src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -174,24 +163,23 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                     getCertificateRequest.getCertificateRequestOptions().getCertificateType();
             // TODO: Should be configurable
             KeyPair keyPair = CertificateStore.newRSAKeyPair(4096);
-            X509Certificate[] caCertificates = getX509CACertificates();
 
             if (certificateType.equals(GetCertificateRequestOptions.CertificateType.SERVER)) {
-                Consumer<X509Certificate> consumer = (t) -> {
+                BiConsumer<X509Certificate, X509Certificate[]> consumer = (serverCert, caCertificates) -> {
                     CertificateUpdateEvent certificateUpdateEvent =
-                            new CertificateUpdateEvent(keyPair, t, caCertificates);
+                            new CertificateUpdateEvent(keyPair, serverCert, caCertificates);
                     getCertificateRequest.getCertificateUpdateConsumer().accept(certificateUpdateEvent);
                 };
                 subscribeToServerCertificateUpdatesNoCSR(getCertificateRequest, keyPair.getPublic(), consumer);
             } else if (certificateType.equals(GetCertificateRequestOptions.CertificateType.CLIENT)) {
-                Consumer<X509Certificate[]> consumer = (t) -> {
+                BiConsumer<X509Certificate, X509Certificate[]> consumer = (clientCert, caCertificates) -> {
```

Comment:
Just in case, this change does not change any external APIs. Subscribers pass a `GetCertificateRequest` which accepts a `CertificateUpdateEvent`, that is still the same. See line 178